packages/base-driver/lib/jsonwp-proxy/proxy.js

```diff
@@ -312,10 +313,31 @@ class JWProxy {
   }
 
   async proxyReqRes (req, res) {
-    const [response, resBodyObj] = await this.proxyCommand(req.originalUrl, req.method, req.body);
+    // ! this method must not throw any exceptions
+    // ! make sure to call res.send before return
+    let statusCode;
+    let resBodyObj;
+    try {
+      let response;
+      [response, resBodyObj] = await this.proxyCommand(req.originalUrl, req.method, req.body);
+      res.headers = response.headers;
+      statusCode = response.statusCode;
+    } catch (err) {
+      [statusCode, resBodyObj] = getResponseForW3CError(
+        errors.isErrorType(err, errors.ProxyRequestError)
+          ? err.getActualError()
+          : err
+      );
+    }
+    res.set('content-type', 'application/json');
```

Comment:
does here need `charset=utf-8` as well?
Probably we provide `application/json; charset=utf-8` regularly.

Reply:
makes sense, fixed